### ISSUE TITLE
Regression in 26.6.0: --offline breaks mock_roles for plain-name (legacy) roles, but not for namespace.collection.role FQCN mocks (introduced by #5066)

### ISSUE THREAD

**Issue by AntoineGGG** [ansible/ansible-lint#5096]
## Summary

Since 26.6.0, running ansible-lint with `--offline` combined with a **plain-name** entry in `mock_roles` (e.g. `my_role`, as opposed to a fully-qualified `namespace.collection.role`) causes that role to no longer be found during syntax-check if a playbook/role actually depends on it, resulting in a fatal `syntax-check` violation:

    the role 'my_role' was not found in <search path>

Mocks declared as `namespace.collection.role` (3-part, collection-style) are **not** affected — they keep working fine under `--offline`. This makes the bug easy to miss: a `.ansible-lint` file that mixes both styles will only break on the plain-name entries.

## Root cause

#5066 ("fix: avoid project cache creation in offline mode") changed `App.__init__` in `src/ansiblelint/app.py`:

```diff
  self.runtime = Runtime(
      project_dir=Path(options.project_dir),
-     isolated=True,
+     isolated=not options.offline,
      require_module=True,
      verbosity=options.verbosity,
  )
```

Before 26.6.0, `Runtime` was always constructed with `isolated=True`, regardless of `--offline`.

`_perform_mockings()` (`src/ansiblelint/_mockings.py`) materializes mocked roles in two different locations depending on the name format:

```python
for role_name in options.mock_roles:
    if re.match(r"\w+\.\w+\.\w+$", role_name):
        # namespace.collection.role -> mocked as a collection role
        path = cache_dir / "collections" / "ansible_collections" / namespace / collection / "roles" / role_dir
    else:
        # plain/legacy role name -> mocked as a bare role
        path = cache_dir / "roles" / role_name
```

These two mock locations are wired into ansible-core's search paths through two different, independently-gated code paths:

1. **Collection-style mocks** (`cache_dir/collections/...`) are added to `collections_paths` by `_add_collections_path_if_needed()`, called **unconditionally** in `get_app()`:
   ```python
   # ansiblelint/app.py, get_app()
   _add_collections_path_if_needed(app.options, app.runtime.config.collections_paths)
   ```
   This path is unaffected by `isolated`/`offline` — collection-format mocks keep working.

2. **Plain-name mocks** (`cache_dir/roles/...`) are only added to `ANSIBLE_ROLES_PATH` inside `ansible_compat.runtime.Runtime._prepare_ansible_paths()`:
   ```python
   alterations_list.extend(
       (
           [
               (roles_path, f"{self.cache_dir}/roles", False),
               (library_paths, f"{self.cache_dir}/modules", False),
               (collections_path, f"{self.cache_dir}/collections", False),
           ]
           if self.isolated
           else []
       ),
   )
   ```
   Since `self.isolated` now equals `not options.offline`, passing `--offline` makes this whole block a no-op — `cache_dir/roles` is silently never added to `ANSIBLE_ROLES_PATH`, so ansible-core can't find the mocked plain-name role anymore.

## Reproduction

`.ansible-lint`:
```yaml
mock_roles:
  - some.collection.role   # will still work
  - some_plain_role        # will break
```

A playbook/role depending on both:
```yaml
roles:
  - role: some.collection.role
  - role: some_plain_role
```

Run:
```
ansible-lint --offline --profile=safety playbook.yml
```

Expected: both roles are mocked, no error.
Actual (26.6.0+): only `some_plain_role` fails —
```
the role 'some_plain_role' was not found in ...
```
`some.collection.role` resolves fine. Removing `--offline` makes the remaining error disappear too, confirming the mocking mechanism itself is intact — only the offline-mode path wiring for plain-name roles is broken.

## Versions

- ansible-lint 26.6.0 (regression), working fine on 26.4.0
- ansible-compat 26.6.0
- ansible-core 2.19.11

## Suggested fix

The `if self.isolated` gate in `_prepare_ansible_paths()` conflates two unrelated concerns: cache *isolation* (where ansible-core's install cache lives) and *mock path injection* (making previously-materialized mock roles discoverable). Injecting `cache_dir/roles` / `cache_dir/modules` into the search paths should not depend on `isolated` — it should happen whenever `_perform_mockings()` actually created mocks there (i.e. `options.mock_roles` contains plain-name entries), mirroring how `_add_collections_path_if_needed()` already does this unconditionally for collection-format mocks.
